### PR TITLE
Refactor Contact class and include tests

### PR DESCRIPTION
**Pull request by nicholastanvis**

### CODE REVIEW

File: src/main/java/seedu/address/model/contact/ContactEmail.java [hunk 1]
```diff
@@ -4,44 +4,50 @@
...
 
-    private static final String SPECIAL_CHARACTERS = "!#$%&'*+/=?`{|}~^.-";
+    private static final String PRINTABLE_CHARACTERS = "!#$%&'*+-/=?^_`{|}~";
     public static final String MESSAGE_CONSTRAINTS = "Emails should be of the format local-part@domain "
             + "and adhere to the following constraints:\n"
             + "1. The local-part should only contain alphanumeric characters and these special characters, excluding "
-            + "the parentheses, (" + SPECIAL_CHARACTERS + ") .\n"
-            + "2. This is followed by a '@' and then a domain name. "
+            + "the parentheses, (" + PRINTABLE_CHARACTERS + ").\n"
+            + "2. The local-part may consist of dots (.) provided that it is not the first character.\n"
```
Comment (goatygoatygoat) [inline src/main/java/seedu/address/model/contact/ContactEmail.java]: provided that it is not the first or last character and provided also that it does not appear consecutively

File: src/main/java/seedu/address/model/contact/ContactPhone.java [hunk 2]
```diff
@@ -4,10 +4,10 @@
...
 /**
- * Represents a Contact's phone number in the address book.
+ * Represents a Contact's phone number in Teaching Assistant.
  * Guarantees: immutable; is valid as declared in {@link #isValidPhone(String)}
  */
-public class Phone {
+public class ContactPhone {
 
 
     public static final String MESSAGE_CONSTRAINTS =
```
Comment (goatygoatygoat) [inline src/main/java/seedu/address/model/contact/ContactPhone.java:13]: #139 Are we following this person's suggestion?

**Review by goatygoatygoat — APPROVED** [review]

File: src/main/java/seedu/address/model/contact/ContactPhone.java (hunk 2 above)
Comment (nicholastanvis) [reply src/main/java/seedu/address/model/contact/ContactPhone.java:13]: sure, will fix

File: src/main/java/seedu/address/model/contact/ContactEmail.java (hunk 1 above)
Comment (nicholastanvis) [reply src/main/java/seedu/address/model/contact/ContactEmail.java]: wip

**Review by lrj689 — APPROVED** [review]

**Review by tsh22 — APPROVED** [review]

**Pull request merged**